### ISSUE TITLE
Failed generating the SQL query in online demo

### ISSUE THREAD

**Issue by 24js** [directus/api#996]
# Bug Report
Failed to generate the SQL query in the online demo app, I don't know if this issue is related to demo only 

## Steps to Reproduce

1. Go to 'https://directus.app/#/collections/sales/91315'
or any other items in the sales collection
2. try increasing Sale Amount field past 10 $
3. See error
```javascript
{code: 9, message: "Failed generating the SQL query.", query: "UPDATE `sales` SET `sale_amount` = '10.23', `id` = '91315' WHERE `id` = '91315'"}
```

**Comment by bjgajjar** [comment]
Hello @24js 

You are getting this error because of the length of `sales_amount` set in the collection. Here the `precision` is 3 and `scale` is 2. So the max value can be 9.99 only! Thus when you try to update the value with `10` it throws the error. Please update that value from the admin setting.

![Issue-996](https://user-images.githubusercontent.com/19200178/58777806-b8761300-85ed-11e9-9fc2-392a064a31ad.png)

**Comment by benhaynes** [comment]
Thanks @bjgajjar — is there a way to catch this error and have it be friendlier (in the App at least, but hopefully the API too)? Perhaps something like: `The value submitted (10) for "Sale Amount" is longer than the field's supported length (3,2). Please submit a shorter value or ask an Admin to increase the length.`

@rijkvanzanten 🔔

**Comment by rijkvanzanten** [comment]
> You are getting this error because of the length of sales_amount set in the collection. Here the precision is 3 and scale is 2. So the max value can be 9.99 only! Thus when you try to update the value with 10 it throws the error. Please update that value from the admin setting.

While this is true, it shouldn't throw a 500 error in that case. Like @benhaynes mentioned, it should throw an error with a description as to why it failed with a 400 status code.

**Comment by bjgajjar** [comment]
Fixed in #997
